Shop_Unity_base/Assets/Scripts/Shop/Model/Customer.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace Model
+{
+    using System;
+    using System.Collections.Generic;
+    using Utility;
+
+    public class Customer
+    {
+        private Inventory inventory;
+        public string Name { get; private set; }
+        public int Gold { get; private set; }
+
+        public Customer(string name)
+        {
+            Name = name;
+            Initialize();
+        }
+
+        public void Initialize()
```

Comment:
Should you be able to reinitialize a customer? Additionally, would it not make sense for Unit Testing to make programmable objects via their constructors so that you don't have to come back to this when it inevitably happens?